src/cli/cmd/fetch.rs

```diff
@@ -0,0 +1,218 @@
+use std::path::{Path, PathBuf};
+use std::process::ExitCode;
+
+use clap::Parser;
+use color_eyre::eyre::{self, WrapErr as _};
+use color_eyre::Result;
+use tokio::fs;
+use tokio::process::Command;
+use tokio::{fs::OpenOptions, io::AsyncWriteExt};
+
+use crate::cli::cmd::nix_command;
+
+use super::{CommandExecute, FlakeHubClient};
+
+/// First line of the error message printed by Nix when --out-link isn't
+/// supported. We use this as a feature test to determine which copying we do.
+const OUT_LINK_NOT_SUPPORTED: &[u8] = b"error: unrecognised flag '--out-link'";
+
+#[derive(Parser)]
+pub(crate) struct FetchSubcommand {
+    /// The FlakeHub flake reference to fetch.
+    /// References must be of this form: {org}/{flake}/{version_req}#{attr_path}
+    flake_ref: String,
+
+    /// Output link to store paths in, a la Nix's `--out-link` option.
+    #[clap(long)]
+    out_link: Option<String>,
```

Comment:
It isn't safe to substitute without a target link, in that the GC can delete it out from under you. I'd prefer an interface where the out link is required.

I'm tempted to call the parameter `target`, and use this as the interface:

`fh fetch flake#ref ./result`

and possibly (but not required in this PR, and arguably we shouldn't) add a way to turn it off: `fh fetch --disable-garbage-collection-safety flake#ref`

Reply:
I implemented it this way to mimic `nix copy` (where `--out-link` is optional), but we can easily require it.